pkg/alertmanager/operator.go

```diff
@@ -808,7 +809,8 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 	a.Status.Conditions = operator.UpdateConditions(a.Status.Conditions, availableCondition, reconciledCondition)
 	a.Status.Paused = a.Spec.Paused
 
-	if _, err = c.mclient.MonitoringV1().Alertmanagers(a.Namespace).UpdateStatus(ctx, a, metav1.UpdateOptions{}); err != nil {
+	aac := applyconfigurationv1.Alertmanager("altertmanager", a.Namespace)
```

Comment:
you need to build a complete *AlertmanagerStatusApplyConfiguration struct to provide to the ApplyStatus() method.

```suggestion
       asac := applyconfigurationv1.AlertmanagerStatus()
       
       ...
       
       asac.WithPaused(a.Spec.Paused)

	aac := applyconfigurationv1.Alertmanager(a.Name, a.Namespace).WithStatus(asac))
```